submissions/5mountains/dom-api-task/js/main.js

```diff
@@ -0,0 +1,102 @@
+const styles = {
+    menuLink: 'menu__link',
+    menuLinkActive: 'menu__link--active',
+    menuItem: 'menu__item',
+    heroItem: 'hero__item',
+    heroItemActive: 'hero__item--active',
+    heroImg: 'hero__image',
+    itemIcon: 'item__icon',
+    itemTitle: 'item__caption',
+    itemPrice: 'item__price',
+    itemInfo: 'item__description'
+};
+const sectionMenu = document.querySelector('.nav__menu');
+const sectionHero = document.querySelector('.hero');
+const url = './data/data.json';
+let selectMenuItem;
+let selectHeroItem;
+
+async function fetchAsync () {
+    const response = await fetch(url);
+    const data = await response.json();
+    return data;
+}   
+  
+fetchAsync()
```

Comment:
In order to increase the readability of your code, let's move this function call to the bottom after all your function declarations.
Also, you can create a new function and place there calling `fetchAsync`. Then add some event listener for `DOMContentLoaded`
```suggestion

function initApp() {
fetchAsync().then(data => {
        const details = createHtml(data);
        drawPage(details);
        toggleTabs();
    })
    .catch(reason => console.log(reason.message));
}
document.addEventListener('DOMContentLoaded', init = () => {
    initApp();
    document.removeEventListener('DOMContentLoaded', init);
});
```

Reply:
Honestly, I'm a little confused about the second part) I should create one more function, put their async function fetchAsync, and then call it in addEventListener('DOMContentLoaded') ? Could you please explain more in details...

Reply:
Now I got it, there was a little misunderstanding from my side! All is clear!